addons/l10n_id_efaktur/models/account_move.py

```diff
@@ -209,9 +209,10 @@ def _generate_efaktur_invoice(self, delimiter):
                     if tax.amount > 0:
                         tax_line += line.price_subtotal * (tax.amount / 100.0)
 
-                invoice_line_unit_price = line.price_unit
-
-                invoice_line_total_price = invoice_line_unit_price * line.quantity
+                # guarantees price to be tax-excluded
+                invoice_line_unit_price = line.tax_ids.compute_all(line.price_unit)['total_excluded']
```

Comment:
You can take price_subtotal and the discount field and calculate the price_subtotal before discount from there and do price_subtotal_before_discount/quantity for the invoice_line_total_price (just be careful when quantity == 0)
That way we avoid the compute_all.  (price_subtotal is always price excluded)  @nni-odoo